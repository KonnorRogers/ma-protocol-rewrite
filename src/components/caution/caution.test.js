import React from "react"
import { render } from "@testing-library/react"
import Caution from "./caution.js"

test("Renders without error", () => {
  const { asFragment } = render(<Caution />)
  expect(asFragment()).toMatchSnapshot()
})

test("Renders with different props", () => {
  const { asFragment, getByTestId } = render(
    <Caution text="My caution" borderColor="green" data-testid="caution" />
  )
  expect(asFragment()).toMatchSnapshot()
  expect(getByTestId("caution")).toBeTruthy()
})
