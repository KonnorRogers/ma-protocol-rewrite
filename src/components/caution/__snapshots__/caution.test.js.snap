// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with different props 1`] = `
<DocumentFragment>
  .emotion-2 {
  margin: auto;
}

.emotion-0 {
  color: var(--link-blue);
  padding: 0.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  margin: 1rem auto -0.25rem auto;
  padding: 0.25rem 1rem;
  text-align: left;
  color: green;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1 {
  margin: 0.75rem auto;
  padding: 1.5rem 2rem;
  border-radius: 12px;
  border: 2px solid green;
  text-align: left;
  font-size: 1rem;
}

@media only screen and (min-width:640px) {
  .emotion-1 {
    font-size: 1.1rem;
  }
}

.emotion-2 {
  margin: auto;
}

.emotion-0 {
  color: var(--link-blue);
  padding: 0.25rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  margin: 1rem auto -0.25rem auto;
  padding: 0.25rem 1rem;
  text-align: left;
  color: green;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1 {
  margin: 0.75rem auto;
  padding: 1.5rem 2rem;
  border-radius: 12px;
  border: 2px solid green;
  text-align: left;
  font-size: 1rem;
}

@media only screen and (min-width:640px) {
  .emotion-1 {
    font-size: 1.1rem;
  }
}

<div
    class="emotion-2"
  >
    <span
      style="display: inline-block;"
    >
      <a
        class="emotion-0"
        href="#caution-1"
      >
        <strong>
          My caution
        </strong>
      </a>
    </span>
    <div
      class="emotion-1"
      data-testid="caution"
      id="caution-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Renders without error 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0.75rem auto;
  padding: 1.5rem 2rem;
  border-radius: 12px;
  border: 2px solid rgba(218,165,32,1);
  text-align: left;
  font-size: 1rem;
}

@media only screen and (min-width:640px) {
  .emotion-0 {
    font-size: 1.1rem;
  }
}

.emotion-0 {
  margin: 0.75rem auto;
  padding: 1.5rem 2rem;
  border-radius: 12px;
  border: 2px solid rgba(218,165,32,1);
  text-align: left;
  font-size: 1rem;
}

@media only screen and (min-width:640px) {
  .emotion-0 {
    font-size: 1.1rem;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
